moira/cache.py

```diff
@@ -1,9 +1,31 @@
 from functools import wraps
 
 from twisted.internet import reactor, defer
+from collections import OrderedDict
 
-CACHE = {}
+#CACHE = {}
+class CacheClass(OrderedDict):
+    """
+    Autocleaning dict.
+    """
+    
+    max_live_time = 2*60**2 # 2 hours
+    
+    def __setitem__(self, key, value, *args, **kwargs):
+        if key in self:
+            del self[key]
+        super(CacheClass, self).__setitem__(key, value, *args, **kwargs)
+    
+    def cleanup(self):
+        now = reactor.seconds()
+        seconds = now - self.max_live_time
+        
+        for key in self.iterkeys():
+            if self[key][0] > seconds:
+                break
+            del self[key]
```

Comment:
I'm not sure if this is a problem, but it's usually a bad idea to delete elements from an iterator while iterating on it.